### PR TITLE
Support of Enter Tower

### PR DESCRIPTION
**Pull request by Akka0**
- allow user select temporary team to enter Tower Dungeon (Abyss Corridor)
- while user exit dungeon, the team will recover

There is still a lot of work to be done.

### CODE REVIEW

File: src/main/java/emu/grasscutter/game/player/TeamManager.java [hunk 1]
```diff
@@ -58,7 +52,13 @@ public class TeamManager {
 	@Transient private final Set<EntityBaseGadget> gadgets;
 	@Transient private final IntSet teamResonances;
 	@Transient private final IntSet teamResonancesConfig;
-	
+
+	private int useTemporarilyTeamIndex = -1;
+	/**
+	 * Temporary Team for tower
+	 */
+	private List<TeamInfo> temporaryTeam;
```
Comment (ghost) [inline src/main/java/emu/grasscutter/game/player/TeamManager.java]: Please add the @Transient annotation to `temporaryTeam` and `useTemporarilyTeamIndex` so they dont get saved to the database.

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/main/java/emu/grasscutter/game/player/TeamManager.java (hunk 1 above)
Comment (Akka0) [reply src/main/java/emu/grasscutter/game/player/TeamManager.java]: Done

**Comment by mcxiaoke** [comment]
maybe a better class name, official dungeon name is **Spiral Abyss**

**Comment by ghost** [comment]
But its called `tower` in the official client code.

**Review by ghost — APPROVED** [review]

**Pull request merged**